onnxruntime/core/providers/cpu/reduction/reduction_ops.h

```diff
@@ -444,6 +454,96 @@ class ReduceAggregatorMax : public ReduceAggregator<T> {
   }
 };
 
+template <>
+class ReduceAggregatorMax<bool> : public ReduceAggregator<bool> {
+ public:
+  inline ReduceAggregatorMax(int64_t N, const bool& init) : ReduceAggregator<bool, bool>(N, init) {}
+
+  static bool aggall(const bool* from_data, int64_t size) {
+    return Eigen::Map<const Eigen::Matrix<bool, Eigen::Dynamic, 1>>(from_data, onnxruntime::narrow<size_t>(size)).cast<int>().maxCoeff();
+  }
+  inline bool aggall(const bool* from_data) {
+    return aggall(from_data, this->N_);
+  }
+  inline void update(const bool& v) { this->accumulator_ = v > this->accumulator_ ? v : this->accumulator_; }
+
+  // Fast reduction
+  static inline FastReduceKind WhichFastReduce() {
+    return FastReduceKind::kKR | FastReduceKind::kRK | FastReduceKind::kKRK | FastReduceKind::kRKR;
+  }
+
+  static void FastReduceKR(const Tensor& input, const gsl::span<const int64_t>& fast_shape,
+                           Tensor& output, concurrency::ThreadPool* tp) {
+    const bool* data = input.Data<bool>();
+    bool* out = output.MutableData<bool>();
+    int64_t stridei = fast_shape[1];
+    concurrency::ThreadPool::TryParallelFor(
+        tp, onnxruntime::narrow<std::ptrdiff_t>(fast_shape[0]), ParallelReduceFastCost(1, stridei, sizeof(bool), 6),
+        [data, stridei, out](std::ptrdiff_t first, std::ptrdiff_t last) {
+          EigenVectorMap<bool>(out + first, last - first) = ConstEigenMatrixMap<bool>(
```

Comment:
instead of duplicating much of the ReduceAggregatorMax/Min code for a `bool` specialization, how about using `if constexpr`?

e.g.,
`if constexpr (std::is_same_v<bool, T>) { /* bool specific impl */ } else { /* generic impl */ }`

Reply:
This is cool. change made according to the suggested.